ex04.ex

```diff
@@ -75,7 +82,14 @@ defmodule Ex04 do
   return value will be the thing you have to manipulate.
   """
 
-  def even_odd . . . "your code"
+  def even_odd([ h | t ]), do: reduce([ h | t ], {[], []},
```

Comment:
## -2

Same comments as above.

Also, using ++ to concatenate lists is really inefficient—it's normally better to use [a|tail] and reverse.

~~~ elixir
  def even_odd(list) do
    list
    |> reverse()
    |> reduce({ [], [] }, &partition_even_odd/2)
  end

  defp partition_even_odd(x, { es, os }) when Integer.is_even(x), do: {[x | es], os}
  defp partition_even_odd(x, { es, os }) when Integer.is_odd(x),  do: {es, [x | os]}
~~~